MdeModulePkg/Universal/DisplayEngineDxe/InputHandler.c

```diff
@@ -752,6 +752,11 @@ GetNumericInput (
         // In Manual input mode, check whether input the negative flag.
         //
         if (Key.UnicodeChar == '-') {
+          if ((NumericOp->Flags & EFI_IFR_DISPLAY) == EFI_IFR_DISPLAY_UINT_DEC){

```

Comment:
What I would do here is simply check for UINT and ManualInput, because this fragment handles two cases:

1. ManualInput and type is INT. `+` and `-` need special treat here for signed integers, and this si what is currently made (before modifications).
2. If not manual input `+` and `-` is used to increase numeric op by the value of `step` attribute. Not all numericvalues have step attribute. Some have step attribute set to 0, which means only ManualInput.

What we need to do here is most likely something like this:

```C
TheKey2:
    switch (Key.UnicodeChar) {

    case '+':
    case '-':
+     if (ManualInput && ((NumericOp->Flags & EFI_IFR_DISPLAY) == EFI_IFR_DISPLAY_UINT_DEC)) {
+       UpdateStatusBar(INPUT_ERROR, TRUE);
+       break;
+     }
      if (ManualInput && IntInput) {
        //
        // In Manual input mode, check whether input the negative flag.
        //
        if (Key.UnicodeChar == '-') {
          if (Negative) {
            break;
          }
          Negative = TRUE;
          PrintCharAt (Column++, Row, Key.UnicodeChar);
        }
      } else {
        if (Key.UnicodeChar == '+') {
          Key.ScanCode = SCAN_RIGHT;
        } else {
          Key.ScanCode = SCAN_LEFT;
        }
        Key.UnicodeChar = CHAR_NULL;
        goto TheKey2;
      }
      break;
```

This should solve it right?

Reply:
Above code snippet tested and QEMU and behaves as intended